Install react and react-dom globally? When installing the cli.

When I clean install the cli I get this weird prompts on the absence of react and react-dom.
When I add them globally it works. Are they set as peer dependencies?

Is this normal? Or am I missing something...

```
npm install -g gatsby-cli
/usr/local/bin/gatsby -> /usr/local/lib/node_modules/gatsby-cli/lib/index.js

> core-js@2.6.10 postinstall /usr/local/lib/node_modules/gatsby-cli/node_modules/core-js
> node postinstall || echo "ignore"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:
> https://opencollective.com/core-js
> https://www.patreon.com/zloirock

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> gatsby-telemetry@1.1.32 postinstall /usr/local/lib/node_modules/gatsby-cli/node_modules/gatsby-telemetry
> node src/postinstall.js


> gatsby-cli@2.8.0 postinstall /usr/local/lib/node_modules/gatsby-cli
> node scripts/postinstall.js

+ gatsby-cli@2.8.0
added 350 packages from 181 contributors in 22.172s


gatsby new myPersonalBlog
/usr/local/lib/node_modules/gatsby-cli/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

Error: Cannot find module 'react-dom'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/gatsby-cli/node_modules/react-redux/lib/utils/reactBatchedUpdates.js:6:17)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)

npm install -g react-dom
npm WARN react-dom@16.10.2 requires a peer of react@^16.0.0 but none is installed. You must install peer dependencies yourself.

+ react-dom@16.10.2
added 7 packages from 3 contributors in 1.537s

gatsby new myPersonalBlog
\/usr/local/lib/node_modules/gatsby-cli/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

Error: Cannot find module 'react'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at /usr/local/lib/node_modules/react-dom/cjs/react-dom.development.js:18:13
    at Object.<anonymous> (/usr/local/lib/node_modules/react-dom/cjs/react-dom.development.js:27925:5)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)

npm install -g react
+ react@16.10.2
added 6 packages from 3 contributors in 0.862s

gatsby new myPersonalBlog
info Creating new site from git: https://github.com/gatsbyjs/gatsby-starter-default.git
Cloning into 'myPersonalBlog'...
```

--- a/e2e-tests/development-runtime/cypress/integration/hot-reloading/new-file.js
+++ b/e2e-tests/development-runtime/cypress/integration/hot-reloading/new-file.js
@@ -1,6 +1,7 @@
 describe(`hot reloading new page component`, () => {
   before(() => {
     cy.exec(`npm run update -- --file src/pages/sample.js`)
+    cy.wait(1000)
   })
 
   beforeEach(() => {

--- a/e2e-tests/development-runtime/cypress/integration/hot-reloading/new-file.js
+++ b/e2e-tests/development-runtime/cypress/integration/hot-reloading/new-file.js
@@ -1,6 +1,9 @@
 describe(`hot reloading new page component`, () => {
   before(() => {
     cy.exec(`npm run update -- --file src/pages/sample.js`)
+    // TO-DO remove `wait` below and fix this properly in core,
+    // we shouldn't have to wait here and core
+    // should be smart enough to recover on it's own.
     cy.wait(1000)
   })
 

--- a/packages/gatsby-cli/package.json
+++ b/packages/gatsby-cli/package.json
@@ -38,6 +38,7 @@
     "progress": "^2.0.3",
     "prompts": "^2.2.1",
     "react": "^16.10.2",
+    "react-dom": "^16.10.2",
     "react-redux": "^7.1.1",
     "redux": "^4.0.4",
     "resolve-cwd": "^2.0.0",

